service/matching/matching_engine.go

```diff
@@ -545,6 +569,42 @@ pollLoop:
 			workerVersionCapabilities: request.WorkerVersionCapabilities,
 			forwardedFrom:             req.GetForwardedSource(),
 		}
+		if e.config.EnableDeployments(req.PollRequest.Namespace) && req.PollRequest.WorkerVersionCapabilities.UseVersioning {
```

Comment:
Deployment workflow is the concern of the physical queue, hence, the best place for this is physicalTaskQueueManager. Basically, physical queue wants to register its deployment when it receives the first poll, before it starts processing that poll or any other polls arriving before the registration is done.

One advantage of that is you don't need to repeat things in different types of polls.